samza-test/src/test/java/org/apache/samza/test/operator/StreamApplicationIntegrationTestHarness.java

```diff
@@ -0,0 +1,245 @@
+ *     createTopic("mytopic", 1);
+ *     produceMessage("mytopic", 0, "key1", "val1");
+ *     runApplication(myApp, "myApp", null);
+ *     List<ConsumerRecord<String, String>> messages = getMessages(outputTopics)
+ *     Assert.assertEquals(messages.size(), 1);
+ *   }
+ * }}</pre>
+ */
+public class StreamApplicationIntegrationTestHarness extends AbstractIntegrationTestHarness {
+  KafkaProducer producer;
+  KafkaConsumer consumer;
+
+  private static final int NUM_EMPTY_POLLS = 2;
+  private static final Duration POLL_TIMEOUT_MS = Duration.ofSeconds(10);
+  private static final String DEFAULT_DESERIALIZER = "org.apache.kafka.common.serialization.StringDeserializer";
+
+  /**
+   * Starts a single kafka broker, and a single embedded zookeeper server in their own threads.
+   * Sub-classes should invoke {@link #zkConnect()} and {@link #bootstrapUrl()}s to
+   * obtain the urls (and ports) of the started zookeeper and kafka broker.
+   */
+  @Override
+  public void setUp() {
+    super.setUp();
+
+    Properties consumerDeserializerProperties = new Properties();
+    consumerDeserializerProperties.setProperty("key.deserializer", DEFAULT_DESERIALIZER);
+    consumerDeserializerProperties.setProperty("value.deserializer", DEFAULT_DESERIALIZER);
+
+    producer = TestUtils.createNewProducer(
+        bootstrapServers(),
+        1,
+        60 * 1000L,
+        1024L * 1024L,
+        0,
+        0L,
+        5 * 1000L,
+        SecurityProtocol.PLAINTEXT,
+        null,
+        Option$.MODULE$.<Properties>apply(new Properties()),
+        new StringSerializer(),
+        new StringSerializer(),
+        Option$.MODULE$.<Properties>apply(new Properties()));
+
+    consumer = TestUtils.createNewConsumer(
+        bootstrapServers(),
+        "group",
+        "earliest",
+        4096L,
+        "org.apache.kafka.clients.consumer.RangeAssignor",
+        30000,
+        SecurityProtocol.PLAINTEXT,
+        Option$.MODULE$.<File>empty(),
+        Option$.MODULE$.<Properties>empty(),
+        Option$.MODULE$.<Properties>apply(consumerDeserializerProperties));
+  }
+
+  /**
+   * Creates a kafka topic with the provided name and the number of partitions
+   * @param topicName the name of the topic
+   * @param numPartitions the number of partitions in the topic
+   */
+  public void createTopic(String topicName, int numPartitions) {
+    TestUtils.createTopic(zkUtils(), topicName, numPartitions, 1, servers(), new Properties());
+  }
+
+  /**
+   * Produces a message to the provided topic partition.
+   * @param topicName the topic to produce messages to
+   * @param partitionId the topic partition to produce messages to
+   * @param key the key in the message
+   * @param val the value in the message
+   */
+  public void produceMessage(String topicName, int partitionId, String key, String val) {
+    producer.send(new ProducerRecord(topicName, partitionId, key, val));
+    producer.flush();
+  }
+
+  @Override
+  public int clusterSize() {
```

Comment:
Should be `public` since it overrides a method from the `KafkaIntegrationTestHarness`. 

All methods of harness classes should be `protected` IMHO. Not sure what the scope for that is. Prefer to keep this `public` for now, and address it in a separate clean-up?